File: hammer-interaction/src/main/java/dev/vini2003/hammer/interaction/registry/client/HINetworking.java
package dev.vini2003.hammer.interaction.registry.client;

import dev.vini2003.hammer.interaction.api.common.manager.InteractionRuleManager;
import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;

import static dev.vini2003.hammer.interaction.registry.common.HINetworking.SYNC_INTERACTION_RULES;

public class HINetworking {
	public static void init() {
		ClientPlayNetworking.registerGlobalReceiver(SYNC_INTERACTION_RULES, new InteractionRuleManager.InteractionRuleSyncHandler());
	}
}

File: hammer-interaction/src/main/kotlin/dev/vini2003/hammer/interaction/registry/client/HINetworking.kt
package dev.vini2003.hammer.interaction.registry.client

import dev.vini2003.hammer.common.packet.Packet
import dev.vini2003.hammer.interaction.common.packet.sync.SyncInteractionRulesPacket
import dev.vini2003.hammer.interaction.registry.common.HINetworking.SYNC_INTERACTION_RULES
import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking

object HINetworking {
	fun init() {
		ClientPlayNetworking.registerGlobalReceiver(SYNC_INTERACTION_RULES, Packet.clientHandler<SyncInteractionRulesPacket>())
	}
}